### PR TITLE
Add support for httpsAgent

### LINKED ISSUE #33: socket hangup error
SO, I am getting this when I connect:
```
{ Error: socket hang up
    at createHangUpError (_http_client.js:313:15)
    at Socket.socketOnEnd (_http_client.js:409:23)
    at Socket.emit (events.js:187:15)
    at endReadableNT (_stream_readable.js:1086:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
  code: 'ECONNRESET',
  config: 
   { adapter: [Function: httpAdapter],
     transformRequest: { '0': [Function: transformRequest] },
     transformResponse: { '0': [Function: transformResponse] },
     timeout: 0,
     xsrfCookieName: 'XSRF-TOKEN',
     xsrfHeaderName: 'X-XSRF-TOKEN',
     maxContentLength: -1,
     validateStatus: [Function: validateStatus],
     headers: 
      { Accept: 'application/json, text/plain, */*',
        'User-Agent': 'axios/0.15.3' },
     method: 'get',
     url: 'http://10.3.0.42:18082/subjects',
     data: undefined },
  response: undefined }
```

My code looks like:
```
var KafkaAvro = require('kafka-avro');
var options = {
	  'group.id': 'kafka',
	  'ssl.ca.location': "./secrets/ca.crt",
	  'ssl.certificate.location': "./secrets/cert.pem",
	  'ssl.key.location': "./secrets/cert-key.pem",
	  'ssl.key.password': "changeit",
	  'security.protocol': "ssl",
	  'metadata.broker.list': "kafka-0.kafka.default:9093"
};

var topicName = 'actionable-event-topic';

var kafkaAvro = new KafkaAvro({
	kafkaBroker: 'kafka-0.kafka.default:9093',
	schemaRegistry: 'http://10.3.0.42:18082'
});

kafkaAvro.init().then(function() {
	console.log("Kafka Avro Ready to go!");
}).catch(err => {
	console.dir(err);
});
```

Not sure if this matters, but I am using SSL on the brokers

### PR DESCRIPTION
**Pull request by ScottChapman**
fixes #33

### CODE REVIEW

**Comment by ricardohbin** [comment]
There are lint erros in your PR :/

**Comment by ScottChapman** [comment]
Aw! ;-) I'll fix shortly.

**Comment by ricardohbin** [comment]
@ScottChapman ok! Can you change the changelog readme with your changes and credit yourself too?

File: package.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {
   "name": "kafka-avro",
-  "version": "1.0.2",
-  "main": "./lib/kafka-avro",
+  "version": "1.0.3",
```
Comment (ricardohbin) [inline package.json]: you dont need to change the dump - it will change when I release to master. Just reset this change and we can release it

**Comment by ricardohbin** [comment]
Merged and published! Thanks

**Pull request merged**